.github/workflows/codeql.yml

```
@@ -0,0 +1,39 @@
+name: CodeQL SAST
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  schedule:
+    - cron: "23 9 * * 3"
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["c#"]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 #v4.2.0
```

```suggestion
        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 #v4.2.0
        with:
          submodules: true
```
    with:
          submodules: true

---

Submodules are required to build Paprika